lib/Doctrine/ORM/Mapping/ClassMetadataInfo.php

```diff
@@ -3310,4 +3316,29 @@ public function getSequencePrefix(AbstractPlatform $platform)
 
         return $sequencePrefix;
     }
+
+    public function applyData($entity, array $data)
+    {
+        foreach ($data as $field => $value) {
+            if ($this->shouldUseData($field, $value)) {
```

Comment:
It can be done, but do you think is needed? I mean, the operations aren't too complex and they're based on attributes that are already caching the information...